crates/bitcoind_rpc/src/lib.rs

```diff
@@ -74,8 +74,12 @@ impl<'c, C: bitcoincore_rpc::RpcApi> Emitter<'c, C> {
     /// tracked UTXO which is confirmed at height `h`, but the receiver has only seen up to block
     /// of height `h-1`, we want to re-emit this transaction until the receiver has seen the block
     /// at height `h`.
-    pub fn mempool(&mut self) -> Result<Vec<(Transaction, u64)>, bitcoincore_rpc::Error> {
+    pub fn mempool<A>(
+        &mut self,
+        expected_unconfirmed_txids: impl IntoIterator<Item = Txid>,
+    ) -> Result<TxUpdate<A>, bitcoincore_rpc::Error> {
```

Comment:
Okay I just realized that returning `TxUpdate` here is a mistake because when we apply `TxUpdate` we don't filter out irrelevant txs.

I think we should do the following:

`mempool()` should return a `MempoolEvent` struct.

```rust
pub struct MempoolEvent {
    pub unseen_txs: Vec<(Transaction, u64)>,
    pub all_txids: HashSet<Txid>,
}
```

We can remove the `expected_unconfirmed_txids` from the `mempool` method as we can use `MempoolEvent::all_txids` to find evicted txs (let's create a helper method on `MempoolEvent` to do this!